File: apps/desktop-client/src/main/java/com/shterneregen/securelan/desktop/service/DefaultRandomNicknameService.java
package com.shterneregen.securelan.desktop.service;

import java.security.SecureRandom;
import java.util.List;
import java.util.Objects;
import java.util.random.RandomGenerator;

public class DefaultRandomNicknameService implements RandomNicknameService {
    private static final List<String> NICKNAMES = List.of(
            "Alice",
            "Bob",
            "Charlie",
            "Diana",
            "Eve",
            "Frank",
            "Grace",
            "Heidi",
            "Ivan",
            "Judy",
            "Mallory",
            "Niaj",
            "Olivia",
            "Peggy",
            "Quentin",
            "Rupert",
            "Sybil",
            "Trent",
            "Uma",
            "Victor",
            "Walter",
            "Yvonne",
            "Zara",
            "Neo",
            "Trinity",
            "Morpheus",
            "Cipher",
            "Echo",
            "Vector",
            "Nova",
            "Orion",
            "Atlas"
    );

    private final RandomGenerator randomGenerator;

    public DefaultRandomNicknameService() {
        this(new SecureRandom());
    }

    public DefaultRandomNicknameService(RandomGenerator randomGenerator) {
        this.randomGenerator = Objects.requireNonNull(randomGenerator, "randomGenerator must not be null");
    }

    @Override
    public String generate() {
        return NICKNAMES.get(randomGenerator.nextInt(NICKNAMES.size()));
    }
}

File: modules/chat-core/src/main/kotlin/com/shterneregen/securelan/chat/service/impl/DefaultRandomNicknameService.kt
package com.shterneregen.securelan.chat.service.impl

import com.shterneregen.securelan.chat.service.RandomNicknameService
import java.security.SecureRandom
import java.util.Objects
import java.util.random.RandomGenerator

class DefaultRandomNicknameService @JvmOverloads constructor(
    randomGenerator: RandomGenerator = SecureRandom(),
) : RandomNicknameService {
    private val randomGenerator: RandomGenerator = Objects.requireNonNull(randomGenerator, "randomGenerator must not be null")

    override fun generate(): String = NICKNAMES[randomGenerator.nextInt(NICKNAMES.size)]

    companion object {
        private val NICKNAMES = listOf(
            "Alice", "Bob", "Charlie", "Diana", "Eve", "Frank", "Grace", "Heidi", "Ivan", "Judy", "Mallory",
            "Niaj", "Olivia", "Peggy", "Quentin", "Rupert", "Sybil", "Trent", "Uma", "Victor", "Walter",
            "Yvonne", "Zara", "Neo", "Trinity", "Morpheus", "Cipher", "Echo", "Vector", "Nova", "Orion", "Atlas",
        )
    }
}
